### ISSUE TITLE
Blocks for user Profiles

### ISSUE THREAD

**Issue by bhubbard** [WordPress/gutenberg#55394]
## What problem does this address?
An easy way to display a users details on the front of the site.

## What is your proposed solution?
Have several blocks to pull a user defaults fields, such as their username, id, bio, social links, etc. 

Also a block query loop thru multiple users on the site.

**Comment by jordesign** [comment]
@bhubbard  Would this be different to the 'Author' block? https://github.com/WordPress/gutenberg/issues/24952 has some suggested enhancements for that.

**Comment by bhubbard** [comment]
@jordesign Yes, this should be a little different from the 'Author' block. Although maybe based on that ticket you linked it could be a partial rename of some blocks. 

As I believe that ticket discusses breaking up the author block into smaller blocks, like one for gravatar, and this I would agree. Although maybe it should be called 'User Gravatar' block? As a author, just happens to be a user who wrote the content for that post/page/etc. If we are not in a loop, these blocks could have a simple dropdown to choose the user for who we want to display the information.

I see one block being a 'User Query' block which brings to the UI query methods from ```wp_user_query```. This would be a block that lets you create patterns using multiple users, such as showing all users with the same role. You might see this common with a 'team' block pattern, or membership site for example.

**Comment by vk17-starlord** [comment]
**Hey @bhubbard @jordesign 👋**

This idea sounds really valuable — I'd love to help work on the **User Profile Blocks**. Before diving in, I’d appreciate a bit of clarity to make sure I'm aligned with your expectations. Here are a few questions to guide the initial implementation:

1. **Do you envision these blocks being independent of the post loop**, with an option to select any registered user manually (via dropdown or similar)?
2. Could you confirm the **initial list of user fields** you'd like blocks for? (e.g., username, ID, gravatar, bio, website, custom meta?)
3. For the proposed **"User Query" block**, should it mirror the behavior of the `Query Loop` block — with role filters, sorting, pagination, etc.?
4. Should these blocks follow any **design or accessibility patterns already established** in similar blocks (e.g., Author block, Query Loop)?
5. Would this feature include support for **custom user meta fields**, or only the default WordPress user fields for now?

Let me know your thoughts — happy to explore and start prototyping based on your vision!

**Comment by bhubbard** [comment]
@vk17-starlord

1. Yes, this should be independent of the post loop.
2. username, First Name, Last Name, email, gravatar, bio, website, should all be on the list. Custom meta would be nice, but doesn't need to make it into the first version of the block.
3. Yes, I think it should mimic the Query Loop as much as possible.
4. Probably, don't know enough to fully answer this question.
5. IT can start without custom user meta fields, although I think long term it will need to support it.

**Comment by flexseth** [comment]
@vk17-starlord - I have a proof of concept that was used as a coding challenge that I can share as a boilerplate. Given user email, it pulls from Gravatar using a NPM package their profile pic and bio. Could be extended to show a URL.

This is all available via the REST API, and it may be helpful to move the package to GitHub from NPM so it can be maintained there. There may have been changes to make this API documented or more available in Core, it's been a couple of years since I created the project. 

If you care to collaborate, we can link up on Slack and I can show you the component and UI structure that exists.

Lemme know if that's helpful!

**Comment by vk17-starlord** [comment]
@flexseth 
– I think the details should ideally be fetched from the WordPress tables via API or WP_Query, rather than using an NPM package. That said, we can definitely connect on the UI structure. Excited to see your approach and have a discussion around this!

**Comment by vk17-starlord** [comment]
@bhubbard - as discussed i have added few individual components in Gutenberg which can be reusecd for user-query-block which I'm currently working on. Wanted to show you the smaller blocks which I have developed

![Image](https://github.com/user-attachments/assets/c13d60bc-ecf8-4fd8-a979-ebaaa458fcb9)

![Image](https://github.com/user-attachments/assets/5aedac04-7875-4477-9fb2-a63dff4f9c37)

**Comment by vk17-starlord** [comment]
@bhubbard - now working on the user-query block which will use this blocks to display results along with layout and styling

**Comment by flexseth** [comment]
> [@flexseth](https://github.com/flexseth) – I think the details should ideally be fetched from the WordPress tables via API or WP_Query, rather than using an NPM package. That said, we can definitely connect on the UI structure. Excited to see your approach and have a discussion around this!

https://docs.gravatar.com/rest/getting-started/

My project is located at https://github.com/flexseth/bioboxfinal

It has a "user switching" feature where you select the user account in the sidebar panel, and it displays the author card.

The card has a round profile image, a heading with the author's name, and a text blurb about their bio that are all pulled from the REST API for Gravatar. I used the NPM package but linked above are the official docs for the Gravatar endpoint.

A user's Gravatar information shows up on their WordPress profile, their Slack profile, at their Gravatar website, and has some components that show up on websites that have comments on them. 

This approach removes extra database fields and removes redundancy. The idea of the Gravatar package is that you can update your information at one place, and it populates everywhere! Maybe you can offer an override if they want to, for example, customize their biography for one website vs. another. Additionally, Gravatar is backed and built by Automattic, so using this existing functionality follows the creed of "Doing things the WordPress way."

Take a look at the scaffolding and component structure in my repo. You will need to install the [Gravatar NPM package](https://www.npmjs.com/package/gravatar) for it to work.

Hopefully that provides some helpful context!

**Comment by vk17-starlord** [comment]
@flexseth  Thank you so much for sharing the detailed context and walkthrough of your implementation — it's super helpful and thoughtful. I really appreciate the clean approach of leveraging Gravatar to reduce redundancy and stay aligned with the WordPress philosophy.

That said, I won’t be able to move forward with this right away. Since this involves integrating a third-party package and affects user profile data display, we’ll need to get feedback and approval from other core contributors before proceeding. It’s important that we run this by the WordPress/Gutenberg core team to ensure it aligns with broader architectural and privacy considerations.

@Mamaduka @joedolson,
Would you be able to take a look at this suggestion when you have some time to review it?
Apologies for the tag — I understand that new blocks aren’t a current priority, but since I’m working on this regardless of priority, I wanted to get your thoughts on the approach.

**Comment by Mamaduka** [comment]
@vk17-starlord, as I mentioned in another issue, new blocks aren’t high priority at the moment.

I think user general user profiles are already in core, anything specialized is probably better suited for separate block plugins.

**Comment by flexseth** [comment]
It looks like there was a [Gravatar block](https://blog.gravatar.com/2024/10/23/upgrade-author-profiles-with-the-gravatar-block-for-wordpress/) introduced late last year by the team at Automattic. It even includes built in metadata for search engines!

**Comment by Mamaduka** [comment]
Thanks, @flexseth! The Gravatar block is a good example. I'm going to close this as "maybe later".
